sirepo/pkcli/admin.py

```diff
@@ -44,6 +50,25 @@ def audit_proprietary_lib_files(*uid):
 
 def create_examples():
     """Adds missing app examples to all users"""
+
+    def _create():
+        for sim_type in feature_config.cfg().sim_types:
+            names = [
+                x.name
+                for x in simulation_db.iterate_simulation_datafiles(
+                    sim_type,
+                    simulation_db.process_simulation_list,
+                    {"simulation.isExample": True},
+                )
+            ]
+            for example in simulation_db.examples(sim_type):
```

Comment:
simulation_db.examples(sim_type) is a constant so cache in a dict outside the call to _access_sim_db_and_callback so you don't incur the cost of an expensive loop invariant for every user. This would be reused below, too.

Reply:
I think simulation_db.examples(sim_type) is going to vary by user. 

I think we need 
```python
import sirepo.auth_db
import sirepo.server

sirepo.server.init()
for d in pkio.sorted_glob(simulation_db.user_path().join("*")):
    if _is_src_dir(d):
        continue
    uid = simulation_db.uid_from_dir_name(d)
    with sirepo.auth_db.session_and_lock(), auth.set_user_outside_of_http_request(
         uid
    ):
```
to get the relevant types to pass into `simulation_db.examples(sim_type)`

otherwise we run into: `AssertionError: no _non_threaded_context`

Reply:
[simulation_db.examples](https://github.com/radiasoft/sirepo/blob/8bf6d7c2fdf28c9979075d305d1988c4484a786a/sirepo/simulation_db.py#L144) does not vary by user. [simulation_db.iterate_simulation_data_files](https://github.com/radiasoft/sirepo/blob/8bf6d7c2fdf28c9979075d305d1988c4484a786a/sirepo/simulation_db.py#L282) does.

Reply:
@robnagler I think I addressed caching of simulation_db.examples lmk what you think